test/e2e/autoscale_test.go

```diff
@@ -201,22 +200,14 @@ func assertScaleUp(ctx *testContext) {
 }
 
 func assertScaleDown(ctx *testContext) {
-	ctx.t.Log("The autoscaler successfully scales down when devoid of traffic. Waiting for scale to zero.")
-	err := pkgTest.WaitForDeploymentState(
-		ctx.clients.KubeClient,
-		ctx.deploymentName,
-		test.DeploymentScaledToZeroFunc,
-		"DeploymentScaledToZero",
-		test.ServingNamespace,
-		autoscaler.DefaultScaleToZeroGracePeriod+autoscaler.DefaultStableWindow+2*time.Minute)
-	if err != nil {
-		ctx.t.Fatalf("Unable to observe the Deployment named %s scaling down. %s", ctx.deploymentName, err)
+	if err := WaitForScaleToZero(ctx.t, ctx.deploymentName, ctx.clients); err != nil {
+		ctx.t.Fatalf("Unable to observe the Deployment named %s scaling down: %v", ctx.deploymentName, err)
 	}
 
 	// Account for the case where scaling up uses all available pods.
 	ctx.t.Log("Wait for all pods to terminate.")
 
-	err = pkgTest.WaitForPodListState(
+	err := pkgTest.WaitForPodListState(
```

Comment:
well if you want to be consistent, you can do `if err :=...; err != nil {}` here as well, despite it's long, it's still readable.